scripts/rollup-plugin-babel-source.js

```diff
@@ -29,13 +29,32 @@ export default function () {
           typeof packageJson["browser"] === "object"
         ) {
           for (const nodeFile in packageJson["browser"]) {
-            const browserFile = packageJson["browser"][nodeFile].replace(
+            const browserFileAsJs = packageJson["browser"][nodeFile].replace(
               /^(\.\/)?lib\//,
               "src/"
             );
-            const nodeFileSrc = path.normalize(
+
+            const browserFileAsTs = browserFileAsJs.replace(/.js$/, ".ts");
+            let browserFile;
+            try {
+              fs.statSync(browserFileAsTs);
+              browserFile = browserFileAsTs;
+            } catch {
+              browserFile = browserFileAsJs;
+            }
```

Comment:
Nit: I realize you copied this logic from the end of this file, but we could simplify it to

```suggestion
            const browserFile = fs.existsSync(browserFileAsTs) ?  browserFileAsTs : browserFileAsJs;
```

Reply:
updated all 3 places to use fs.existsSync